feat: added a ball, added bouncing

Solpaw opened the pull request:
Póki co piłeczka buguje się gdy uderza o bok paletki.

Solpaw committed:
feat: added a ball, added bouncing

Solpaw committed:
Fix: ball no longer bugs out on paddle corners

Solpaw committed:
fix: deleted obsolete logic

jponiatowski commented on js/Ball.js:
```diff
@@ -5,14 +5,45 @@ class Ball {
+            location.reload(true);          
+        }
+        // Od paletki
+        let paddleY = this.cvs.height-paddle.paddleHeight;
```
**const** paddleY

jponiatowski commented on js/Ball.js:
```diff
@@ -5,14 +5,45 @@ class Ball {
+        }
+        // Od paletki
+        let paddleY = this.cvs.height-paddle.paddleHeight;
+        let distX = Math.abs(ballX - (paddle.paddleX + paddle.paddleWidth/2));
```
**const** distX

jponiatowski commented on js/Ball.js:
```diff
@@ -5,14 +5,45 @@ class Ball {
+        // Od paletki
+        let paddleY = this.cvs.height-paddle.paddleHeight;
+        let distX = Math.abs(ballX - (paddle.paddleX + paddle.paddleWidth/2));
+        let distY = Math.abs(ballY - (paddleY + paddle.paddleHeight/2));
```
**const** distY

jponiatowski commented on js/Ball.js:
```diff
@@ -5,14 +5,45 @@ class Ball {
+        if (distX <= paddle.paddleWidth/2 && distY <= paddle.paddleHeight/2 + this.ballRadius){
+            dy = -dy;
+        }
+        let pitaX = distX - paddle.paddleWidth/2;
```
**const** pitaX

jponiatowski commented on js/Ball.js:
```diff
@@ -5,14 +5,45 @@ class Ball {
+            dy = -dy;
+        }
+        let pitaX = distX - paddle.paddleWidth/2;
+        let pitaY = distY - paddle.paddleHeight/2;
```
**const** pitaY

jponiatowski commented on js/Ball.js:
```diff
@@ -5,14 +5,45 @@ class Ball {
+        if (ballX <= this.ballRadius || this.cvs.width-ballX <= this.ballRadius){
+            dx = -dx;
+        }
+        // Od górnej ściany
```
Trochę sobie skomplikowałeś to odbijanie w pionie, a z tą paletką to już wgl :D 

 if (ballY + dy < this.ballRadius) {
            dy = -dy;
        } else if (ballY + dy > this.cvs.height - this.ballRadius) {
            if (ballX > paddleX && x < paddleX + paddleWidth) {
                dy = -dy;
            } else {
                alert((`Game over! Your score: ${score}`);;
                window.clearInterval(interval);
                location.reload(true); 
            }
        }  
    };

Solpaw replied:
Zmienić to?
Pomyślałem, że mam zrobić odbijanie od paletki, a nie ściany w miejscu paletki.

jponiatowski commented on js/Ball.js:
```diff
@@ -5,14 +5,45 @@ class Ball {
     };
 
-    bouncing(){
+    bouncing(ballX, ballY, paddle){
```
aż się prosi żeby sobie zdestrukturyzować to paddle
bouncing(ballX, ballY. { paddleX, paddleWidth })

jponiatowski commented on js/index.js:
```diff
@@ -19,14 +19,21 @@ let y = cvs.height-30; // początkowa pozycja piłeczki oś y
     document.querySelector("#score span").innerText = score;
 };
 
+const moveBall = () => {
```
jak już to przenieś tą funkcję nad funkcję start, bo teraz ją wywołujesz przed jej deklaracją

Solpaw committed:
fix: requested changes

Changed let to const where needed, deconstructed paddle parameter, moved moveBall() up
